ex/high_throughput_inference/mock_app.py

```diff
@@ -183,19 +155,32 @@ def name(self):
 
     parser = argparse.ArgumentParser("Mock application")
     parser.add_argument("--device", default="cpu")
+    parser.add_argument("--log_max_batchsize", default=8, type=int)
     args = parser.parse_args()
 
     resnet = ResNetWrapper("resnet50", f"resnet50.{args.device.upper()}.pt")
 
     client = ProtoClient(timing_on=True)
     client.set_model(resnet.name, resnet.model)
 
-    total_iterations = 100
+    pt_model = torch.jit.load(io.BytesIO(initial_bytes=(resnet.model))).to("cuda:0")
+
+    TOTAL_ITERATIONS = 100
 
-    for batch_size in [1, 2, 4, 8, 16, 32, 64, 128]:
-        logger.info(f"Batch size: {batch_size}")
-        for iteration_number in range(total_iterations + int(batch_size==1)):
+    for log2_bsize in range(args.log_max_batchsize+1):
+        b_size: int = 2**log2_bsize
+        logger.info(f"Batch size: {b_size}")
+        for iteration_number in range(TOTAL_ITERATIONS + int(b_size==1)):
             logger.info(f"Iteration: {iteration_number}")
-            client.run_model(resnet.name, resnet.get_batch(batch_size))
+            batch = resnet.get_batch(b_size)
+            remote_result = client.run_model(resnet.name, batch)
+            logger.info(client._perf_timer.get_last("total_time"))
```

Comment:
Consider if we can simplify the perf timer code to avoid needing to track total time by adding a perf timer context here. Does this achieve your goal? 

```
batch = resnet.get_batch(b_size)

perf_timer.start("driver")
remote_result = client.run_model(resnet.name, batch)
perf_timer.stop()
```

both options may be unnecessary if `client.run_model` were to start a perf timer to track the total time?

Reply:
Do you mean another `PerfTimer` instance? I thought it would be easier to keep all measurements in the same object (and dump them as a single numpy array)